feature/gribi/mpls_in_udp/otg_tests/mpls_in_udp_scale/mpls_in_udp_scale_test.go

```diff
@@ -0,0 +1,1680 @@
+					WithOperationType(constants.Delete).
+					AsResult(),
+			)
+
+			idx++
+		}
+	}
+
+	return adds, dels
+}
+
+func staticARPWithUniversalIP(t *testing.T, dut *ondatra.DUTDevice, vrfsList, portList []string, sDstIP, sRoutePrfLen, sDstMac, protoType string, indx int) {
+	t.Helper()
+	sb := new(gnmi.SetBatch)
+	for v, vrfName := range vrfsList {
+		protoPath := gnmi.OC().NetworkInstance(vrfName).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+		proto := &oc.NetworkInstance_Protocol{Identifier: oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, Name: ygot.String(deviations.StaticProtocolName(dut))}
+		gnmi.BatchUpdate(sb, protoPath.Config(), proto)
+
+		sp := protoPath.Static(sDstIP + sRoutePrfLen)
+
+		for i, portName := range portList {
+			nhIndex := fmt.Sprintf("%d", indx+v*len(portList)+i)
+			port := dut.Port(t, portName)
+
+			// Create next-hop entry
+			nh := &oc.NetworkInstance_Protocol_Static_NextHop{Index: ygot.String(nhIndex), InterfaceRef: &oc.NetworkInstance_Protocol_Static_NextHop_InterfaceRef{Interface: ygot.String(port.Name())}}
+
+			// Add next-hop config under the static route
+			gnmi.BatchUpdate(sb, sp.NextHop(nhIndex).Config(), nh)
+
+			// Add static ARP entry for that interface
+			gnmi.BatchUpdate(sb, gnmi.OC().Interface(port.Name()).Config(), configStaticArp(port.Name(), sDstIP, sDstMac, protoType))
+
+			t.Logf("Added static route in VRF %s: %s -> %s (NextHop %s)", vrfName, sDstIP, port.Name(), nhIndex)
+		}
+	}
+	sb.Set(t, dut)
+}
+
+// validateMPLSPacketCapture analyzes a packet capture on the given OTG port and validates MPLS-in-UDP encapsulation against the expected parameters. Returns an error if validation fails.
+func validateMPLSPacketCapture(t *testing.T, ate *ondatra.ATEDevice, otgPortName string, pr *packetResult, labelList []uint64) error {
+	t.Helper()
+	t.Logf("=== PACKET CAPTURE VALIDATION START for port %s ===", otgPortName)
+
+	// Get raw packet bytes
+	packetBytes := ate.OTG().GetCapture(t, gosnappi.NewCaptureRequest().SetPortName(otgPortName))
+	if len(packetBytes) == 0 {
+		return fmt.Errorf("no packet data captured on port %s", otgPortName)
+	}
+	t.Logf("Captured %d bytes from port %s", len(packetBytes), otgPortName)
+
+	// Write capture to temporary pcap file
+	tmpFile, err := os.CreateTemp("", "*.pcap")
+	if err != nil {
+		return fmt.Errorf("could not create temporary pcap file: %v", err)
+	}
+	if _, err := tmpFile.Write(packetBytes); err != nil {
+		return fmt.Errorf("could not write packet data: %v", err)
+	}
+	tmpFile.Close()
+
+	handle, err := pcap.OpenOffline(tmpFile.Name())
+	if err != nil {
+		return fmt.Errorf("could not open pcap file: %v", err)
+	}
+	defer handle.Close()
+
+	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
+
+	// Optimize label lookups by using a map
+	labelMap := make(map[uint64]struct{}, len(labelList))
+	for _, l := range labelList {
+		labelMap[l] = struct{}{}
+	}
```

Comment:
Using `bool` instead of an empty struct is more common and reads much cleaner.

Also, please don't repeat var type in name - https://google.github.io/styleguide/go/decisions#variable-name-vs-type

```suggestion
    seenLabels := make(map[uint64]bool, len(labelList))
	for _, l := range labelList {
		seenLabels[l] = true
	}
```